Core.Tests/Overlay/WebsocketBroadcastServerTest.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Core.Overlay;
+using Microsoft.Extensions.Logging.Abstractions;
+using NUnit.Framework;
+
+namespace Core.Tests.Overlay
+{
+    [Category("IntegrationTest")]
+    public class WebsocketBroadcastServerTest
+    {
+        private const int Port = 53427;
+        private Task _wsServerListen = null!;
+        private WebsocketBroadcastServer _server = null!;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _server = new WebsocketBroadcastServer(NullLogger<WebsocketBroadcastServer>.Instance, "localhost", Port);
+            _wsServerListen = _server.Listen();
+        }
+
+        [TearDown]
+        public async Task TearDown()
+        {
+            await _server.Stop();
+            await _wsServerListen;
+        }
+
+        private static async Task<WebsocketMessageStreamClient> CreateConnectedClient()
+        {
+            var url = new Uri($"ws://localhost:{Port}");
+            var wsClient = new WebsocketMessageStreamClient();
+            await wsClient.Connect(url, CancellationToken.None);
+            await Task.Delay(TimeSpan.FromMilliseconds(10)); // wait for the server to accept the connection
```

Comment:
10ms seems pretty arbitrary, are you sure it'll be enough? I can see it leading to failed tests in the future